nipype/utils/filemanip.py

```diff
@@ -684,36 +685,37 @@ def loadpkl(infile):
 
     unpkl = None
     with indirectory(infile.parent):
-        pkl_file = pklopen(infile.name, 'rb')
-
-        try:  # Look if pkl file contains version file
-            pkl_metadata_line = pkl_file.readline()
-            pkl_metadata = json.loads(pkl_metadata_line)
-        except (UnicodeDecodeError, json.JSONDecodeError):
-            # Could not get version info
-            pkl_file.seek(0)
-
-        try:
-            unpkl = pickle.load(pkl_file)
-        except UnicodeDecodeError:
-            # Was this pickle created with Python 2.x?
-            unpkl = pickle.load(pkl_file, fix_imports=True, encoding='utf-8')
-            fmlogger.info('Successfully loaded pkl in compatibility mode.')
-        # Unpickling problems
-        except Exception as e:
-            if pkl_metadata and 'version' in pkl_metadata:
-                from nipype import __version__ as version
-                if pkl_metadata['version'] != version:
-                    fmlogger.error("""\
-Attempted to open a results file generated by Nipype version %s, \
-with an incompatible Nipype version (%s)""", pkl_metadata['version'], version)
-                    raise e
-            fmlogger.error("""\
-No metadata was found in the pkl file. Make sure you are currently using \
-the same Nipype version from the generated pkl.""")
-            raise e
-        finally:
-            pkl_file.close()
+        with SoftFileLock(infile.name + '.lock'):
```

Comment:
Maybe best acquire + release in the finally, in this case?

Reply:
the finally is w.r.t an exception inside the `context`